tests/test_helper.go

```diff
@@ -0,0 +1,249 @@
+		},
+		Spec: migapi.MigMigrationSpec{
+			MigPlanRef: &core.ObjectReference{
+				Namespace: MigrationNamespace,
+				Name:      planRef,
+			},
+			Stage:       stage,
+			QuiescePods: quiesce,
+		},
+	}
+}
+
+func NewMigPlan(namespaces []string, name string) *migapi.MigPlan {
+	return &migapi.MigPlan{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: MigrationNamespace,
+		},
+		Spec: migapi.MigPlanSpec{
+			Namespaces: namespaces,
+			SrcMigClusterRef: &core.ObjectReference{
+				Namespace: MigrationNamespace,
+				Name:      E2ETestObjectName,
+			},
+			DestMigClusterRef: &core.ObjectReference{
+				Namespace: MigrationNamespace,
+				Name:      TestDestinationCluster,
+			},
+			MigStorageRef: &core.ObjectReference{
+				Name:      E2ETestObjectName,
+				Namespace: MigrationNamespace,
+			},
+		},
+	}
+}
+
+func NewMigStorage() (*migapi.MigStorage, *core.Secret) {
+	return &migapi.MigStorage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      E2ETestObjectName,
+				Namespace: MigrationNamespace,
+			},
+			Spec: migapi.MigStorageSpec{
+				BackupStorageConfig: migapi.BackupStorageConfig{
+					CredsSecretRef: &core.ObjectReference{
+						Namespace: ConfigNamespace,
+						Name:      TestStorageSecret,
+					},
+					AwsBucketName: os.Getenv(AWSBUCKETNAME),
+				},
+				BackupStorageProvider: os.Getenv(BACKUPSTORAGEPROVIDER),
+				// TODO: define env variable. can these both be different?
+				VolumeSnapshotProvider: os.Getenv(BACKUPSTORAGEPROVIDER),
+				VolumeSnapshotConfig: migapi.VolumeSnapshotConfig{
+					CredsSecretRef: &core.ObjectReference{
+						Namespace: ConfigNamespace,
+						Name:      TestStorageSecret,
+					},
+				},
+			},
+		}, &core.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      TestStorageSecret,
+				Namespace: ConfigNamespace,
+			},
+			Type: TestSecretType,
+			Data: map[string][]byte{
+				"aws-access-key-id":     []byte(os.Getenv(AWSACCESSKEY)),
+				"aws-secret-access-key": []byte(os.Getenv(AWSSECRETKEY)),
+			},
+		}
+}
+
+func NewMigCluster(saToken []byte) (*migapi.MigCluster, *core.Secret) {
+	return &migapi.MigCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      E2ETestObjectName,
+				Namespace: MigrationNamespace,
+			},
+			Spec: migapi.MigClusterSpec{
+				IsHostCluster: false,
+				URL:           os.Getenv(SOURCEURL),
+				ServiceAccountSecretRef: &core.ObjectReference{
+					Namespace: ConfigNamespace,
+					Name:      TestClusterSecret,
+				},
+				Insecure:            true,
+				ExposedRegistryPath: os.Getenv(EXPOSEDREGISTRYPATH),
+			},
+		}, &core.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      TestClusterSecret,
+				Namespace: ConfigNamespace,
+			},
+			Data: map[string][]byte{
+				"saToken": saToken,
+			},
+			Type: TestSecretType,
+		}
+}
+
+// We are assuming that the controller CR is created and controller is running
+func NewMigrationController(install_ui bool, install_controller bool) *unstructured.Unstructured {
```

Comment:
camel case, please?

Go programmer don't like snake case :P haha

Reply:
Oh, Okay! Didn't know this :rofl: